src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestCaseBuildingPreparer.java

```diff
@@ -17,22 +15,30 @@
 import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
 import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
 import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.HandlerUtils;
 import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacaseapi.domain.service.DueDateService;
 
 @Component
 public class RequestCaseBuildingPreparer implements PreSubmitCallbackHandler<AsylumCase> {
 
     private final int legalRepresentativeBuildCaseDueInDays;
     private final int legalRepresentativeBuildCaseDueFromSubmissionDate;
+    private final int legalRepresentativeBuildCaseDueInDaysAda;
     private final DateProvider dateProvider;
+    private final DueDateService dueDateService;
 
     public RequestCaseBuildingPreparer(
             @Value("${legalRepresentativeBuildCase.dueInDays}") int legalRepresentativeBuildCaseDueInDays,
             @Value("${legalRepresentativeBuildCase.dueInDaysFromSubmissionDate}") int legalRepresentativeBuildCaseDueFromSubmissionDate,
+            @Value("${legalRepresentativeBuildCase.dueInDaysAda}") int legalRepresentativeBuildCaseDueInDaysAda,
```

Comment:
Following the existing naming convention: legalRepresentativeBuildCaseAda.dueInDays would be better here. Second object is to determine the event in specific. But only if you are putting another commit, add this change.